src/typeahead/typeahead.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  Directive,
+  OnInit,
+  Input,
+  ComponentRef,
+  ComponentFactoryResolver,
+  ViewContainerRef,
+  Injector,
+  Renderer,
+  ElementRef,
+  TemplateRef
+} from '@angular/core';
+import {Observable} from 'rxjs/Rx';
+import 'rxjs/add/operator/filter';
+import 'rxjs/add/operator/let';
+import {Positioning} from '../util/positioning';
+import {NgbTypeaheadWindow, ResultTplCtx} from './typeahead-window';
+import {PopupService} from '../util/popup';
+import {NgModel} from '@angular/common';
+import {isString} from '../util/util';
+import {NgbTypeaheadValueAccessor} from './typeahead-valueaccessor';
+
+/**
+ * NgbTypeahead directive provides a simple way of creating powerful typeaheads from any text input
+ */
+@Directive({
+  selector: 'input[ngbTypeahead][ngModel]',
+  host: {
+    '[class.open]': '_windowRef !== null',
+    '(document:click)': '_closePopup()',
+    '(keydown.arrowDown)': '_onKeyDown($event)',
+    '(keydown.arrowUp)': '_onKeyUp($event)',
+    '(keydown.enter)': '_onKeyEnter()',
+    '(keydown.esc)': '_closePopup()'
```

Comment:
I think that we should have a dedicated method here (`_onKeyEsc` ?) so we could stop propagation of the event. This is needed for cases where typeahead would be placed inside a modal and not stopping propagation would close a modal as well. At least this is my theory :-)